lib/puppet/agent.rb

```diff
@@ -49,6 +49,7 @@ def run(client_options = {})
             return
           rescue StandardError => detail
             Puppet.log_exception(detail, _("Could not run %{client_class}: %{detail}") % { client_class: client_class, detail: detail })
+            1
```

Comment:
I think what was originally intended was to return nil, as we do in the rescue that precedes this one. Then at https://github.com/puppetlabs/puppet/blob/master/lib/puppet/application/agent.rb#L368-L369 if no exit status was returned we exit 1. That being said, this works fine. Seems more explicit.